### DISCUSSION TITLE
How to set NullValueHandling to ignore globally?

### DISCUSSION THREAD

**Discussion by dfaruque** [serenity-is/Serenity#6085] [Help] [answered]
It seems that when a dialog is loads the entity, the properties with `null ` value also transferred from the server. 
in old serenity (.net 45), the properties with `null` values are not serialized. How can I achieve the same behavior in serenity (.net 5)?

eg.
in old serenity : 
`this.entity.someField` is `undefined` if the value is `null` in the DB.
but in new serenity:
`this.entity.someField` is `null`  if the value is `null` in the DB.

I like the old behavior. it will transfer fewer data.

Thanks.

**Comment by VictorTomaili** [comment, marked as answer]
if you set the value as null in dialog (front-end), probably you want to save this value as null in the database. 
if you want to update partial form (for example 2 fields for a row) all other extra rows must be non touched. and if you want to set one of the fields as null from this example (2 fields for a row) how can you determine which fields must be null or non touched?

For this reason, all form data must be serialized. It's doesn't matter if it's null or not. 

Current behavior
| Database | Form (Serialized) | Result |
| ----------- | ----------- | ----------- |
| Id (1) | [hidden] Id | 1 |
| Name (John) | John | John |
| Surname (Doe) | Null | Null (expected update) |
| Age (20) | (not presented in form) | 20 (nothing changed) |


With null value handling
| Database | Form (Serialized) | Result |
| ----------- | ----------- | ----------- |
| Id (1) | [hidden] Id | 1 |
| Name (John) | John | John |
| Surname (Doe) |  | Null (expected update) |
| Age (20) | (not presented in form) | Null (unwanted update) |

**Reply by dfaruque** [reply]
Thanks for your great explanation.

I agree that all form data must be serialized. But I am talking about retrieving data from the server while loading the entity in the dialog. 

Suppose, we have a form with 100 properties, and only 3 of them have the value in the database.
Currently, when the form is loaded from the server, it transferred all the 100 properties include `null` value properties. but I think only the 3 properties with non-null values should be transferred. and we can simply assign null to the form fields which is undefined in the retrieved entity.

**Reply by VictorTomaili** [reply]
You can edit JSON options in the `Startup.cs` file for null value handling. Probably there is no side effect when you remove null values from responses.
